src/ext/java/org/opentripplanner/ext/flex/edgetype/FlexTripEdge.java

```diff
@@ -56,7 +56,7 @@ public State traverse(State s0) {
     editor.incrementTimeInSeconds(timeInSeconds);
     editor.incrementWeight(timeInSeconds);
     editor.resetEnteredNoThroughTrafficArea();
-    return editor.makeState();
+    return State.ofNullable(editor.makeState());
```

Comment:
Almost every makeState() call is wrapped inside State.ofNullable(), which kind of looks like unnecessary 
 boilerplate. Should this repeating wrapper be provided by stateEditor? E.g. editor.makeStateArray()?

Reply:
That's a really good suggestion which I will implement.

Reply:
done in https://github.com/opentripplanner/OpenTripPlanner/pull/4841/commits/060be6e3078341305e8e84aa6a0323510677d64e